dhall-yaml/src/Dhall/Yaml.hs

```diff
@@ -51,26 +52,22 @@ jsonToYaml
     -> Bool
     -> ByteString
 jsonToYaml json documents quoted =
+  let encoder = Data.YAML.Aeson.encodeValue' schemaEncoder YT.UTF8 in
   case (documents, json) of
     (True, Data.Aeson.Array elems)
       -> Data.ByteString.intercalate "\n---\n"
-         $ fmap (Data.ByteString.Lazy.toStrict. (Data.YAML.Aeson.encodeValue' schemaEncoder YT.UTF8). (:[]))
-         $ Data.Vector.toList elems
-    _ -> Data.ByteString.Lazy.toStrict (Data.YAML.Aeson.encodeValue' schemaEncoder YT.UTF8 [json])
+         $ (Data.ByteString.Lazy.toStrict . encoder . (:[])) <$> Data.Vector.toList elems
+    _ -> Data.ByteString.Lazy.toStrict (encoder [json])
   where
     style (Y.SStr s)
         | "\n" `Text.isInfixOf` s =
             Right (YE.untagged, YE.Literal YE.Clip YE.IndentAuto, s)
         | quoted || Text.all isNumberOrDateRelated s || isBoolString =
             Right (YE.untagged, YE.SingleQuoted, s)
       where
-        isBoolString
-          | Text.length s > 5 = False
```

Comment:
This check was to prevent doing `Text.toLower` unnecessarily.

Reply:
Is `Text.toLower` a really expensive computation?

Reply:
Not for most **keys**, but this could potentially be a very long value. I benchmarked some common kubernetes payloads with large base64-encoded strings, and the difference when not lowercasing those was noticeable.

Reply:
Ok I see. I'll keep that check then